src/objects/core/zcl_abapgit_serialize.clas.abap

```diff
@@ -224,7 +225,13 @@ CLASS zcl_abapgit_serialize IMPLEMENTATION.
       iv_language         = io_dot_abapgit->get_main_language( )
       ii_log              = ii_log
       iv_force_sequential = lv_force ).
-    APPEND LINES OF lt_found TO ct_files.
+* transfer one by one and delete from original table as APPEND LINES OF
+* leads to doubling memory usage during execution of statement which
+* could result in SYSTEM_NO_ROLL dump
+    LOOP AT lt_found ASSIGNING <ls_found>.
+      APPEND <ls_found> TO ct_files.
+      DELETE lt_found.
```

Comment:
suggest making the INDEX explicit in this statement

Reply:
I was actually trained in WDF to do DELETE in Loop without any addition if the current line should be delete. It was expected to be the most robust (no need to check if index is fetched at correct location, no struggling with index vs. tabix) and and fastest ("itab knows its current line"). 

As I struggled to find any evidence on SAP help, I tested this out... and the reality is that DELETE ... INDEX is actually always about 20 % faster (tested on NW 7.00 ... up to S4HANA 2020). 

Will have to look through some own code now.... 😥

Reply:
yea... I think its better to always specify the index explicitly

Before this commit, I was not sure what would happen, does APPEND set the implicit index, or is it taken from the LOOP? Anyhow, now the developer can read what happens without knowing ABAP internal mechanisms

Reply:
For sake of completeness, I just found the proper documentation:

https://help.sap.com/doc/abapdocu_750_index_htm/7.50/en-US/abapdelete_itab_line.htm#!ABAP_ONE_ADD@2@
"This variant is only possible within a LOOP across the same internal table. The current table row of the LOOP is then deleted implicitly."

As per my best understanding (and what I have learnt in WDF), it always deletes the record that is currently processed by LOOP. It does not rely on any system field (index vs. tabix). So it works as expected, but impact on performance still makes me a little sad..